ut/s3_head_object_action_test.cc

```diff
@@ -258,3 +262,126 @@ TEST_F(S3HeadObjectActionTest, SendSuccessResponse) {
 
   action_under_test->send_response_to_s3_client();
 }
+
+TEST_F(S3HeadObjectActionTest, SendSuccessResponseWhenVerEnabled) {
+  CREATE_OBJECT_METADATA;
+  std::string versioning_status = "Enabled";
+  EXPECT_CALL(*(bucket_meta_factory->mock_bucket_metadata),
+              get_bucket_versioning_status())
+      .Times(AtLeast(1))
+      .WillRepeatedly(ReturnRef(versioning_status));
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata), get_state())
+      .WillOnce(Return(S3ObjectMetadataState::present));
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata),
+              get_last_modified_gmt())
+      .WillOnce(Return("Sunday, 29 January 2017 08:05:01 GMT"));
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata),
+              get_content_length_str()).WillOnce(Return("512"));
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata), get_md5())
+      .Times(AtLeast(1))
+      .WillOnce(Return("abcd1234abcd"));
+
+  std::map<std::string, std::string> meta_map{{"key", "value"}};
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata),
+              get_user_attributes())
+      .Times(1)
+      .WillOnce(ReturnRef(meta_map));
+
+  EXPECT_CALL(*mock_request, set_out_header_value(_, _)).Times(AtLeast(7));
+  EXPECT_CALL(*mock_request, send_response(S3HttpSuccess200, _))
+      .Times(AtLeast(1));
+
+  action_under_test->send_response_to_s3_client();
+}
+
+TEST_F(S3HeadObjectActionTest, validateObjInfoWithDelMarker) {
+  CREATE_OBJECT_METADATA;
+  action_under_test->object_metadata =
+      object_meta_factory->mock_object_metadata;
+  EXPECT_CALL(*(object_meta_factory->mock_object_metadata), is_delete_marker())
+      .WillOnce(Return(true));
+  EXPECT_CALL(*(mock_request), has_query_param_key("versionId"))
+      .WillRepeatedly(Return(false));
+  EXPECT_CALL(*mock_request, set_out_header_value(_, _)).Times(AtLeast(2));
+  action_under_test->validate_object_info();
+}
```

Comment:
We can additional checks for error code. Like - 
  EXPECT_STREQ("NoSuchKey",
               action_under_test->get_s3_error_code().c_str());

In case versionId is pesent in the request - 
  EXPECT_STREQ("MethodNotAllowed",
               action_under_test->get_s3_error_code().c_str());

Reply:
done